### DISCUSSION TITLE
Instrumentation plugin

### DISCUSSION THREAD

**Discussion by vjpr** [swc-project/swc#4513] [Ideas]
There are a few use cases for instrumenting source code:

- code coverage
- runtime debugging (like wallaby.js does by showing the values of expressions at runtime inline in your editor)
- visualization (showing graphical representation of control flow / data structures / etc.)

Has anyone started on a plugin that does instrumentation?

Maybe an instrumentation api should be exposed by swc.

## Discussions

https://github.com/vercel/next.js/discussions/30174#discussioncomment-1522635
https://github.com/vercel/next.js/discussions/30529
https://twitter.com/acemarke/status/1428465251062779907
https://github.com/swc-project/swc/issues/4114

**Comment by kwonoj** [comment]
Dupe to https://github.com/swc-project/swc/issues/4114 , you may not need repost already existing issue .
